File: abc/abc310/b.py
n, m = map(int, input().split())
p, f = list(), list()
for _ in range(n):
    inputs = list(map(int, input().split()))
    p.append(inputs[0])
    f.append(set(inputs[2:]))

satisfied = False

for i, bad in enumerate(zip(p, f)):
    for j, good in enumerate(zip(p, f)):

        if i == j:
            continue

        bad_price, bad_funcs = bad
        good_price, good_funcs = good

        if good_price > bad_price:
            continue

        if good_price == bad_price and good_funcs > bad_funcs:
            satisfied = True
        if good_price < bad_price and good_funcs >= bad_funcs:
            satisfied = True

print('Yes' if satisfied else 'No')

File: abc/abc310/b.cpp
#include <algorithm>
#include <iostream>
#include <set>
#include <vector>

int main() {
    int n, m;
    std::cin >> n >> m;

    std::vector<int> p(n);
    std::vector<std::set<int>> f(n);
    for (int i = 0; i < n; i++) {
        std::cin >> p[i];
        int c;
        std::cin >> c;
        std::set<int> func;
        for (int j = 0; j < c; j++) {
            int input;
            std::cin >> input;
            func.insert(input);
        }
        f[i] = func;
    }

    bool satisfied = false;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (i == j) {
                continue;
            }

            if (p[i] < p[j]) {
                continue;
            }

            if (f[i].size() > f[j].size()) {
                continue;
            }

            std::set<int> res;
            std::set_intersection(f[i].begin(), f[i].end(), f[j].begin(), f[j].end(), std::inserter(res, res.end()));
            if (res != f[i]) {
                continue;
            }

            if (p[i] == p[j] && f[i].size() == f[j].size()) {
                continue;
            }

            satisfied = true;
        }
    }

    std::cout << (satisfied ? "Yes" : "No") << std::endl;
    return 0;
}